Add Contact Intial Design Implementation

Vinay18Mishra opened the pull request:
I have done with the below listed items on current PR:

UX: 4.3.2.0 - UI without visuals and Saving functionality working fine.
UX: 4.3.2.2 - UI without visuals complete , Use temporary wifi Icon Image, functionality on progress
UX: 4.3.2.3 - UI without visuals complete, Functionality is on progress.

@Ashish-Developer : Please review.

Vinay18Mishra committed:
Add Contact : Intial Design Implementation

harman-jenkins commented:
Harman-Jenkins: pr_review_failed
**Commit Message errors:**
- 1st line doesn't follow the [template](http://stteamsite.symphonyteleca.com/redsquare/Wiki/Commit_message_template.aspx): Bug 123456 - Short description 

**Lint errors:**
jshint check:
apps/communications/contacts/js/contacts.js: line 75, col 132, Line is too long. (ERROR)
apps/communications/contacts/js/contacts.js: line 503, col 4, Unnecessary semicolon. (ERROR)

2 errors
There were errors while linting the files, please see above.
Please read https://github.com/mozilla-b2g/gaia/tree/master/build/jshint/README.md for more information.

harman-jenkins commented:
Harman-Jenkins: pr_review_failed
**Commit Message errors:**
- 1st line doesn't follow the [template](http://stteamsite.symphonyteleca.com/redsquare/Wiki/Commit_message_template.aspx): Bug 123456 - Short description 

**Lint errors:**
jshint check:

0 errors
:( -1 errors has been added to apps/communications/contacts/style/app.css.
Errors in /export/jenkins/jobs/Gaia_PR_checker/workspace/checkstyle/gaia/apps/communications/contacts/style/app.css
margin-top can't be used with display: inline. line 472, col 3
Warnings in /export/jenkins/jobs/Gaia_PR_checker/workspace/checkstyle/gaia/apps/communications/contacts/style/app.css
Negative text-indent doesn't work well with RTL. If you use text-indent for image replacement explicitly set direction for that item to ltr. line 125, col 3
Use of !important line 531, col 3
Use of !important line 532, col 3
Use of !important line 536, col 3
Use of !important line 571, col 3
Use of !important line 653, col 3

Warnings in /export/jenkins/jobs/Gaia_PR_checker/workspace/checkstyle/gaia/shared/style/contacts/contacts_buttons.css
Use of !important line 3, col 3
Use of !important line 7, col 3
Use of !important line 11, col 3
Use of !important line 18, col 3
Use of !important line 35, col 3
Use of !important line 139, col 3
Use of !important line 143, col 3

There were errors while linting the files, please see above.

harman-jenkins commented:
Harman-Jenkins: pr_review_failed
**Commit Message errors:**
- 1st line doesn't follow the [template](http://stteamsite.symphonyteleca.com/redsquare/Wiki/Commit_message_template.aspx): Bug 123456 - Short description 

**Lint errors:**
jshint check:

0 errors
:( -1 errors has been added to apps/communications/contacts/style/app.css.
Errors in /export/jenkins/jobs/Gaia_PR_checker/workspace/checkstyle/gaia/apps/communications/contacts/style/app.css
margin-top can't be used with display: inline. line 472, col 3
Warnings in /export/jenkins/jobs/Gaia_PR_checker/workspace/checkstyle/gaia/apps/communications/contacts/style/app.css
Negative text-indent doesn't work well with RTL. If you use text-indent for image replacement explicitly set direction for that item to ltr. line 125, col 3
Use of !important line 531, col 3
Use of !important line 532, col 3
Use of !important line 536, col 3
Use of !important line 571, col 3
Use of !important line 653, col 3

Warnings in /export/jenkins/jobs/Gaia_PR_checker/workspace/checkstyle/gaia/shared/style/contacts/contacts_buttons.css
Use of !important line 3, col 3
Use of !important line 7, col 3
Use of !important line 11, col 3
Use of !important line 18, col 3
Use of !important line 35, col 3
Use of !important line 139, col 3
Use of !important line 143, col 3

There were errors while linting the files, please see above.

Vinay18Mishra committed:
Add Contact : Fix the UI and Functionality of Phone, Email and Compnay Screen

harman-jenkins commented:
Harman-Jenkins: pr_review_failed
**Commit Message errors:**
- 1st line doesn't follow the [template](http://stteamsite.symphonyteleca.com/redsquare/Wiki/Commit_message_template.aspx): Bug 123456 - Short description 

**Lint errors:**
jshint check:
apps/communications/contacts/js/contacts.js: line 552, col 4, Missing semicolon. (ERROR)
apps/communications/contacts/js/contacts.js: line 567, col 103, Line is too long. (ERROR)
apps/communications/contacts/js/contacts.js: line 568, col 106, Line is too long. (ERROR)
apps/communications/contacts/js/contacts.js: line 571, col 4, Missing semicolon. (ERROR)
apps/communications/contacts/js/contacts_tag.js: line 73, col 9, 'contacts' is not defined. (ERROR)
apps/communications/contacts/js/contacts_tag.js: line 75, col 9, 'contacts' is not defined. (ERROR)
apps/communications/contacts/js/views/form.js: line 28, col 7, 'setCurrentTag' is defined but never used. (ERROR)
apps/communications/contacts/js/views/form.js: line 28, col 22, 'setCurrentElement' is defined but never used. (ERROR)
apps/communications/contacts/js/views/form.js: line 28, col 41, 'setCurrentEvent' is defined but never used. (ERROR)

9 errors
There were errors while linting the files, please see above.
Please read https://github.com/mozilla-b2g/gaia/tree/master/build/jshint/README.md for more information.

Vinay18Mishra committed:
Add Contact : Fix JSLint Errors

Vinay18Mishra commented:
1. apps/communications/contacts/js/contacts.js: line 552, col 4, Missing semicolon. : Fixed
2. apps/communications/contacts/js/contacts.js: line 567, col 103, Line is too long.  : Fixed
3. apps/communications/contacts/js/contacts.js: line 568, col 106, Line is too long.  : Fixed
4. apps/communications/contacts/js/contacts.js: line 571, col 4, Missing semicolon. : Fixed

Below jslint issues are not valid, because it's use global in communications. Then please ignore this:
1. apps/communications/contacts/js/contacts_tag.js: line 73, col 9, 'contacts' is not defined. (ERROR)
2. apps/communications/contacts/js/contacts_tag.js: line 75, col 9, 'contacts' is not defined. (ERROR)
3. apps/communications/contacts/js/views/form.js: line 28, col 7, 'setCurrentTag' is defined but never used. 
4. apps/communications/contacts/js/views/form.js: line 28, col 22, 'setCurrentElement' is defined but never used. 
5. apps/communications/contacts/js/views/form.js: line 28, col 41, 'setCurrentEvent' is defined but never used. 

@Ashish-Developer : please review

harman-jenkins commented:
Harman-Jenkins: pr_review_failed
**Commit Message errors:**
- 1st line doesn't follow the [template](http://stteamsite.symphonyteleca.com/redsquare/Wiki/Commit_message_template.aspx): Bug 123456 - Short description 

**Lint errors:**
jshint check:
apps/communications/contacts/js/contacts.js: line 573, col 4, Missing semicolon. (ERROR)
apps/communications/contacts/js/contacts_tag.js: line 73, col 9, 'contacts' is not defined. (ERROR)
apps/communications/contacts/js/contacts_tag.js: line 75, col 9, 'contacts' is not defined. (ERROR)
apps/communications/contacts/js/views/form.js: line 28, col 7, 'setCurrentTag' is defined but never used. (ERROR)
apps/communications/contacts/js/views/form.js: line 28, col 22, 'setCurrentElement' is defined but never used. (ERROR)
apps/communications/contacts/js/views/form.js: line 28, col 41, 'setCurrentEvent' is defined but never used. (ERROR)

6 errors
There were errors while linting the files, please see above.
Please read https://github.com/mozilla-b2g/gaia/tree/master/build/jshint/README.md for more information.

TatianaMigacheva commented:
@Vinay18Mishra 
Please pay attention on "commit message" in future commits.

Also I have some instructions for you regarding integration of Contacts.
PR#28 for Contact application has been merged into Feature branch: bug-1214767-contacts.
bug-1214767-contacts is intended to accumulate all Harman commits for [Bug#1214767](https://bugzilla.mozilla.org/show_bug.cgi?id=1214767). Our rules about PR target branch have been changed because of Mozilla is stopping all activities for v2.2r. We just want to accumulate all features on separate branches for doing final patches for Mozilla. So please help us to complete Contact integration.

The next step for your is to rebase your dev branch on [bug-1214767-contacts](https://github.com/harman-red-square/gaia/tree/bug-1214767-contacts).
Below is steps how to do rebase:

1) squash existing commits into one (get <sha1> for this commit). 
Currently you have 3 commits for Contact so you will run:

```
$git rebase HEAD~3
```

get <sha1> for new commit

2) checkout from bug-1214767-contacts 

```
git fetch upstream
git checkout -b <your dev branch> upstream/bug-1214767-contacts
```

3) cherry-pick <sha1> on  bug-1214767-contacts.

```
git cherry-pick  <sha1> 
//sha1 is from step 1.
```

4) Open new PR against bug-1214767-contacts (this PR#36 can be closed with commenting the new PR number)

All further Contacts PR (if any) should be opened against bug-1214767-contacts branch.

Please ping me via Link if any issues. Thanks!

TatianaMigacheva commented:
PR is closed because PR#40 is correct one.
